src/main.ts

```diff
@@ -151,6 +159,44 @@ export class Purchases {
     return Purchases.getSharedInstance();
   }
 
+  /**
+   * Initializes the Purchases SDK by prefetching some optional data.
+   * Currently only prefetching branding information. You can call this method
+   * after configuring the SDK to speed up the first call to
+   * {@link Purchases.purchase}.
+   */
+  public async initialize(): Promise<void> {
+    if (this.isInitialized()) {
+      Logger.verboseLog("Purchases is already initialized. Skipping.");
+      return;
+    }
+    if (this._initializationPromise !== null) {
```

Comment:
In a multithreading context, this would be to be much better handled, but JS is single threaded, so it shouldn't cause any issues I believe?

Reply:
correct, shouldn't unless we're running different promises in parallel which I don't think we're doing anywhere